wolfcrypt/src/evp.c

```diff
@@ -4123,6 +4161,11 @@ int wolfSSL_EVP_MD_type(const WOLFSSL_EVP_MD *md)
                 ctx->gcmDecryptBuffer = NULL;
             }
             ctx->gcmDecryptBufferLen = 0;
+            if (ctx->gcmAuthIn) {
+                XFREE(ctx->gcmAuthIn, NULL, DYNAMIC_TYPE_OPENSSL);
+                ctx->gcmAuthIn = NULL;
+            }
+            ctx->gcmAuthInSz = 0;
```

Comment:
Does reuse of the cipher reset the gcmAuthInSz?

Reply:
If you do not call EVP_CIPHER_CTX_free then the same AAD is used for the next Init-Update-Final cycle.